### PR TITLE
Fix potential duplicated slash in pidfile path

### PR DESCRIPTION
**Pull request by tristan957**
Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

File: lib/kvdb/kvdb_home.c [hunk 1]
```diff
@@ -115,7 +115,7 @@ kvdb_home_pidfile_path_get(const char *home, char *buf, const size_t buf_sz)
 
     int n;
 
-    n = snprintf(buf, buf_sz, "%s/" PIDFILE_NAME, home);
+    n = snprintf(buf, buf_sz, "%s%s" PIDFILE_NAME, home, home[strlen(home) - 1] == '/' ? "" : "/");
```
Comment (alexttx) [inline lib/kvdb/kvdb_home.c:118]: If you really want to clean up dups, you might need to loop on last char and backup until you're not sitting at a `/`.

Comment (alexttx) [reply lib/kvdb/kvdb_home.c:118]: But since we can't easily eliminate dups in the middle, maybe we shouldn't worry about dups anywhere in path.

**Review by alexttx — APPROVED** [review]

File: lib/kvdb/kvdb_home.c (hunk 1 above)
Comment (tristan957) [reply lib/kvdb/kvdb_home.c:118]: I have a different PR for removing dup slashes like that in the works. Plan is to handle it at the kvdb_interface level.

Comment (beckerg) [reply lib/kvdb/kvdb_home.c:118]: Do we disallow relative paths in kvdb_home?

**Review by beckerg — APPROVED** [review]

File: lib/kvdb/kvdb_home.c (hunk 1 above)
Comment (tristan957) [reply lib/kvdb/kvdb_home.c:118]: No we don't, but absolute paths are stored or mclass paths don't sit within the home

**Pull request merged**